packages/ui/src/components/va-radio/VaRadio.vue

```diff
@@ -1,116 +1,183 @@
+        <slot>
+          {{ option.label || option.value }}
+        </slot>
+      </span>
+    </label>
+  </VaMessageListWrapper>
 </template>
 
 <script lang="ts">
-import { defineComponent, PropType, computed } from 'vue'
+import { defineComponent, PropType, computed, shallowRef } from 'vue'
 
 import { generateUniqueId } from '../../utils/uuid'
-import { useComponentPresetProp, useColors, useFormFieldProps, useFormField } from '../../composables'
+import {
+  useComponentPresetProp,
+  useColors,
+  useSelectable,
+  Elements,
+  useSelectableProps,
+  useSelectableEmits,
+} from '../../composables'
+import { VaMessageListWrapper } from '../va-input'
+import type { option } from './types'
 
 export default defineComponent({
   name: 'VaRadio',
-  emits: ['update:modelValue', 'focus'],
+  components: { VaMessageListWrapper },
+  emits: useSelectableEmits,
   props: {
-    ...useFormFieldProps,
+    ...useSelectableProps,
     ...useComponentPresetProp,
-    modelValue: { type: [Boolean, Array, String, Object, Number] as PropType<boolean | null | string | number | Record<any, unknown> | unknown[]>, default: null },
-    option: { type: [String, Boolean, Object, Number] as PropType<any>, default: null },
+    modelValue: {
+      type: [Boolean, Array, String, Object, Number] as PropType<
+        option['value']
+      >,
+      default: null,
+    },
+    options: {
+      type: Array<option>,
+      default: () => [],
+    },
     name: { type: String, default: '' },
-    label: { type: String, default: '' },
     leftLabel: { type: Boolean, default: false },
     color: { type: String, default: 'primary' },
-    tabindex: { type: [String, Number], default: 0 },
+    ariaLabel: { type: String, default: undefined },
   },
   setup (props, { emit }) {
     const { getColor } = useColors()
+    const elements: Elements = {
+      container: shallowRef<HTMLElement>(),
+      input: shallowRef<HTMLElement>(),
+      label: shallowRef<HTMLElement>(),
+    }
 
-    const isActive = computed(() => props.modelValue === props.option)
-
-    const { computedClasses } = useFormField('va-radio', props)
+    const {
+      isChecked,
+      computedError,
+      computedErrorMessages,
+      validationAriaAttributes,
+      onBlur,
+      onFocus,
+    } = useSelectable(props, emit, elements)
 
     const computedClass = computed(() => ({
       'va-radio--left-label': props.leftLabel,
-      ...computedClasses,
+      'va-radio--selected': isChecked.value,
+      'va-radio--readonly': props.readonly,
+      'va-radio--disabled': props.disabled,
+      'va-radio--indeterminate': props.indeterminate,
+      'va-radio--error': computedError.value,
     }))
 
+    const selectOption = (option: option['value']) => {
+      emit('update:modelValue', option)
+    }
+
+    const labelStyle = computed(() => {
+      return {
+        color: computedError.value ? getColor('danger') : '',
+        padding: props.leftLabel ? '0 0.5rem 0 0' : '0 0 0 0.5rem',
```

Comment:
Would be nice to move it to CSS variable.

```suggestion
        padding: props.leftLabel ? '0 var(--va-radio-label-padding) 0 0' : '0 0 0 var(--va-radio-label-padding)',
```